components/clp-package-utils/clp_package_utils/scripts/compress.py

```diff
@@ -21,28 +25,165 @@
     args_parser.add_argument(
         "-t", "--tags", help="A comma-separated list of tags to apply to the compressed archives."
     )
+    args_parser.add_argument(
+        "--no-progress-reporting", action="store_true", help="Disables progress reporting."
+    )
+
+
+def _validate_fs_input_args(
+    parsed_args: argparse.Namespace,
+    args_parser: argparse.ArgumentParser,
+) -> None:
+    # Validate some input paths were specified
+    if len(parsed_args.paths) == 0 and parsed_args.path_list is None:
+        args_parser.error("No paths specified.")
+
+    # Validate paths were specified using only one method
+    if len(parsed_args.paths) > 0 and parsed_args.path_list is not None:
+        args_parser.error("Paths cannot be specified on the command line AND through a file.")
+
+
+def _validate_s3_input_args(
+    parsed_args: argparse.Namespace, args_parser: argparse.ArgumentParser, clp_config: CLPConfig
+) -> None:
+    if StorageEngine.CLP_S != clp_config.package.storage_engine:
+        raise ValueError(
+            f"input type {InputType.S3} is only supported for the storage engine {StorageEngine.CLP_S}."
+        )
+
+    # Validate aws credentials were specified using only one method
+    aws_credential_file = parsed_args.aws_credentials_file
+    aws_access_key_id = parsed_args.aws_access_key_id
+    aws_secret_access_key = parsed_args.aws_secret_access_key
+    if aws_credential_file is not None:
+        if not pathlib.Path(aws_credential_file).exists():
+            raise ValueError(f"credentials file {aws_credential_file} doesn't exist.")
+
+        if aws_access_key_id is not None or aws_secret_access_key is not None:
+            args_parser.error(
+                "aws_credentials_file can not be specified together with aws_access_key_id or aws_secret_access_key."
+            )
+
+    elif bool(aws_access_key_id) != bool(aws_secret_access_key):
+        args_parser.error(
+            "aws_access_key_id and aws_secret_access_key must be both specified or left unspecified."
+        )
+
+
+def main(argv):
+    clp_home = get_clp_home()
+    default_config_file_path = clp_home / CLP_DEFAULT_CONFIG_FILE_RELATIVE_PATH
+
+    args_parser = argparse.ArgumentParser(description="Compresses files from filesystem/s3")
+    input_type_args_parser = args_parser.add_subparsers(dest="input_type")
+
+    fs_compressor_parser = input_type_args_parser.add_parser(InputType.FS)
+    _add_common_arguments(fs_compressor_parser, default_config_file_path)
+    fs_compressor_parser.add_argument("paths", metavar="PATH", nargs="*", help="Paths to compress.")
+    fs_compressor_parser.add_argument(
+        "-f", "--path-list", dest="path_list", help="A file listing all paths to compress."
+    )
+
+    s3_compressor_parser = input_type_args_parser.add_parser(InputType.S3)
+    _add_common_arguments(s3_compressor_parser, default_config_file_path)
+    s3_compressor_parser.add_argument("url", metavar="URL", help="URL of object to be compressed")
```

Comment:
```suggestion
    s3_compressor_parser.add_argument("url", metavar="URL", help="URL of the object to be compressed")
```

Reply:
How about URL of objects to be compressed?
Since the URL can be pointing to a path prefix